### PR TITLE
TRA-4202 role management

### PR DESCRIPTION
**Pull request by RamiBerm**

### CODE REVIEW

File: agent/kratos/kratos.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ selfservice:
 log:
   level: info
   format: text
-  leak_sensitive_values: true
+  leak_sensitive_values: false
```
Comment (IgorGov) [inline agent/kratos/kratos.yml:60]: is it related to this change?

File: agent/pkg/api/socket_routes.go [hunk 2]
```diff
@@ -49,7 +48,7 @@ func init() {
 func WebSocketRoutes(app *gin.Engine, eventHandlers EventHandlers, startTime int64) {
 	app.GET("/ws", func(c *gin.Context) {
 		websocketHandler(c.Writer, c.Request, eventHandlers, false, startTime)
-	}, middlewares.RequiresAuth())
+	})
```
Comment (IgorGov) [inline agent/pkg/api/socket_routes.go:51]: is it related to bug that client doesn't send the token in custom header? 
how did it worked before?

File: agent/pkg/routes/config_routes.go [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
+
+	"github.com/gin-gonic/gin"
 )
 
 func ConfigRoutes(ginApp *gin.Engine) {
 	routeGroup := ginApp.Group("/config")
 	routeGroup.Use(middlewares.RequiresAuth())
 
-	routeGroup.POST("/tapConfig", controllers.PostTapConfig)
+	routeGroup.POST("/tapConfig", middlewares.RequiresAdmin(), controllers.PostTapConfig)
```
Comment (IgorGov) [inline agent/pkg/routes/config_routes.go:14]: should we rename it to just "tap"? so there will be no duplicate "config" in endpoint:
/config/tapConfig -> /config/tap

Comment (IgorGov) [reply agent/pkg/routes/config_routes.go:14]: another thought we should rename it to tappedNamespace (we have different "tap" endpoints some for tapPods status, some for tapper status..)

File: agent/kratos/kratos.yml (hunk 1 above)
Comment (RamiBerm) [reply agent/kratos/kratos.yml:60]: no but I noticed this is set to true and we probably dont want it that way

File: agent/pkg/routes/config_routes.go (hunk 3 above)
Comment (RamiBerm) [reply agent/pkg/routes/config_routes.go:14]: renamed to `/tap`

File: agent/pkg/providers/user_role_provider.go [hunk 4]
```diff
@@ -0,0 +1,180 @@
+package providers
```
Comment (IgorGov) [inline agent/pkg/providers/user_role_provider.go:1]: so we won't end up with a package with too many providers, I suggest start splitting them:
"user role provider"
"user provide"

File: agent/pkg/providers/install_provider.go [hunk 5]
```diff
@@ -16,3 +21,27 @@ func IsInstallNeeded() (bool, error) {
 		return !anyUserExists, nil
 	}
 }
+
+func CreateAdminUser(password string, ctx context.Context) (token *string, err error, formErrorMessages map[string][]ory.UiText) {
```
Comment (IgorGov) [inline agent/pkg/providers/install_provider.go:25]: can we move this file content to user provider?

File: agent/pkg/providers/user_provider.go [hunk 6]
```diff
@@ -66,17 +66,17 @@ func PerformLogin(username string, password string, ctx context.Context) (*strin
 	return result.SessionToken, nil
 }
 
-func VerifyToken(token string, ctx context.Context) (bool, error) {
+func VerifyToken(token string, ctx context.Context) (*ory.Session, error) {
```
Comment (IgorGov) [inline agent/pkg/providers/user_provider.go:69]: can we rename the package name to "user provider"?

File: agent/pkg/controllers/install_controller.go [hunk 7]
```diff
@@ -16,3 +16,8 @@ func IsSetupNecessary(c *gin.Context) {
 		c.JSON(http.StatusOK, IsInstallNeeded)
 	}
 }
+
+func SetupAdminUser(c *gin.Context) {
+	token, err, formErrorMessages := providers.CreateAdminUser(c.PostForm("password"), c.Request.Context())
```
Comment (IgorGov) [inline agent/pkg/controllers/install_controller.go:21]:
```suggestion
	token, err, formErrorMessages := userProvider.CreateAdmin(c.PostForm("password"), c.Request.Context())
```

Comment (IgorGov) [reply agent/pkg/controllers/install_controller.go:21]: see the comment for splitting to dedicated providers below..

**Review by IgorGov — APPROVED** [review]

**Pull request merged**